docs/tutorials/query-language.md

```diff
@@ -53,6 +53,58 @@ let peopleWhoseNameContainsA = realm.objects('Contact').filtered('name CONTAINS[
 let Johns = realm.objects('Contact').filtered('name ==[c] "john"');
 ```
 
+You can do simple wildcard matching with `LIKE` which supports using `?` to match a single character and `*` to match zero or multiple characters.
+
+Example:
+```JS
+// Matches "John" and "Johnny"
+let Johns = realm.objects('Contact').filtered('name LIKE "John*"');
+```
+
 ### Composition
 Use parentheses and the `&&`/`AND` and `||`/`OR` operators to compose queries. You can negate a predicate with `!`/`NOT`.
 
+### Queries on collections
+
+When objects contain lists you can query into them using the collection operators `ANY`, `ALL` and `NONE`.
+
+Example:
+```JS
+// Find contacts with one or more teenage friends
+let teens = realm.objects('Contact').filtered('ANY friends.age < 14');
+
+// Find contacts where all friends are older than 21
+let adults = realm.objects('Contact').filtered('ALL friends.age > 21');
+```
+
+You can query on aggregates over properties in the lists using the aggregate operators `.@count`, `.@avg`, `.@min`, `.@max` and `.@sum`.
+
+Example:
+```JS
+// Find contacts without friends
+let teens = realm.objects('Contact').filtered('friends.@count == 0');
+
+// Find contacts where the average age of their friends is above 40
+let adults = realm.objects('Contact').filtered('friends.@avg.age > 40"');
+```
+
+Subqueries using the `SUBQUERY` operator allows you to filter the lists across multiple parameters while querying them.
+
+Example:
+```JS
+// Find contacts with friends above 21 in SF
+let teens = realm.objects('Contact').filtered('SUBQUERY(friends, $friend, $friend.age > 21 AND $friend.city = 'SF').@count > 0');
+```
+
+### Backlink queries
+
+Other objects can link to an object and you can query on that releationship using the `@links` and `@links.ClassName.PropertyName` syntax:
```

Comment:
`@links` by itself is not currently supported, it must be proceeded by a class name and property name. Is this something that would return the sum of all backlink columns? If that is desired, let's open a core issue to track it.

Additionally, the class name part must be prefixed by "class_" which exposes something that should be kept internal. I would actually prefer that we keep backlink queries undocumented if possible, since we will have support for querying over named backlinks in https://github.com/realm/realm-js/pull/1660

Reply:
I'll create an issue for standalone links queries.

>  would actually prefer that we keep backlink queries undocumented if possible, since we will have support for querying over named backlinks 

In js it is common to open realms without specifying a schema, relying on the dynamic nature of the language, so you cannot rely on there being named backlinks defined. Could we filter out the "class_" prefix in the parser?

Reply:
Yes, it should be simple to try to match the exact class name and if that fails add a "class_" prefix automatically.

Reply:
It might be enough if we just add an option to auto-add the prefix that can be enabled when bindings use the parser (since the "class_" prefix is never exposed at the user level).